include/clad/Differentiator/Differentiator.h

```diff
@@ -128,6 +128,30 @@ inline CUDA_HOST_DEVICE unsigned int GetLength(const char* code) {
     return f(static_cast<Args>(args)...);
   }
 
+#ifdef __CUDACC__
+  template <bool EnablePadding, class... Rest, class F, class... Args,
+            class... fArgTypes,
+            typename std::enable_if<EnablePadding, bool>::type = true>
+  CUDA_HOST_DEVICE void
+  execute_with_default_args(list<Rest...>, F f, list<fArgTypes...>, dim3 grid,
+                            dim3 block, size_t shared_mem, cudaStream_t stream,
```

Comment:
```suggestion
                            dim3 block, size_t shared_mem, cudaStream_t stream,
```
Can we wrap that in a CUDA-specific macro, say CUDA_ARGS, which gets expanded only in cuda mode.

This will allow you to wrap the cudaLaunchKernel and friends with `#ifdef __CUDACC__` and allow duplicating the overly complicated templates already...

Reply:
Alternatively, we can go even deeper and assume that in cuda mode `Args` has `size_t shared_mem, cudaStream_t stream` and extract it before forwarding the rest.

Reply:
Done